Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2221,6 +2225,44 @@ namespace Babylon
                 }
             }
 
+            Napi::Value FillPoses(const Napi::CallbackInfo& info)
+            {
+                const auto spaces = info[0].As<Napi::Array>();
+                auto transforms = info[2].As<Napi::Float32Array>();
+                if (spaces.Length() != (transforms.ElementLength() >> 4))
+                {
+                    throw std::runtime_error{"Number of spaces doesn't match number of transforms * 16."};
+                }
+
+                for (uint32_t spaceIdx = 0; spaceIdx < spaces.Length(); spaceIdx++)
+                {
+                    const auto& jointSpace = *spaces[spaceIdx].As<Napi::External<xr::System::Session::Frame::Space>>().Data();
+                    const auto transformMatrix = CreateTransformMatrix(jointSpace, false);
+                    memcpy(transforms.Data() + (spaceIdx << 4), transformMatrix.data(), sizeof(float) * 16);
```

Comment:
sorry for the late comment

nit: inconsistency between `<< 4` and `* 16`

and also std::memcpy would be better